woocommerce/woocommerce-ios
# Dashboard: sync stats after swiping to another time range tab

Closes: #7639 


- [x] ⚠️ Please review https://github.com/woocommerce/woocommerce-ios/pull/7638 first, which is the base branch for this PR

### Description


There are two types of gestures that the user can change the time range tab on the My store tab for the store stats: tapping on the time range tab, or swiping to the previous/next tab. Tapping on the tab is probably the most common gesture, because the issue with swiping hadn't been reported until when I was testing a separate PR https://github.com/woocommerce/woocommerce-ios/pull/7638. This PR fixes the issue where the stats aren't synced when swiping to another time range tab.

The proposed fix was to observe all tab selection (either by tapping or swiping) using `PagerTabStripIsProgressiveDelegate.updateIndicator(for:fromIndex:toIndex:withProgressPercentage:indexWasChanged:)`, and replace the previous observable `isSelectedTimeRangeReady: Bool` with `selectedTimeRangeIndex: Int?`. The initial tab is still set after `loadLastTimeRange` in `viewDidLoad`, and ignored in the first `PagerTabStripIsProgressiveDelegate.updateIndicator` callback during UI initialization. `isSyncing` boolean was also replaced by a set of syncing tabs to allow syncing multiple tabs at the same time, which is common when swiping/tapping on another tab quickly.

### Testing instructions


- Launch the app, log in if needed --> the last selected tab should be shown on the My store tab
- Swipe to a different time range tab --> the target tab should always sync stats for the first time (see the previous broken behavior in the screencast in https://github.com/woocommerce/woocommerce-ios/issues/7639). also, stats API requests should only be made for the visited tabs (initial tab + any selected tabs)
- Tap to a different time range tab --> the target tab should always sync stats for the first time as before. also, stats API requests should only be made for the visited tabs (initial tab + any selected tabs)
---
- [x] @jaclync tests the steps in https://github.com/woocommerce/woocommerce-ios/pull/7638 to make sure the persisted tab behavior stays the same


---
